ImGuiFileDialog.cpp

```diff
@@ -1,72 +1,100 @@
-#include "ImGuiFileDialog.h"
+// From https://github.com/aiekick/ImGuiFileDialog
+//
+// Modified by AlastairGrowcott@yahoo.com to work on Posix systems.
+
+
+#include <sys/types.h>
+#include <dirent.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+
+#include "imgui_file_dlg.h"
 #include "imgui.h"
-#include "dirent/dirent.h"
 
 #define IMGUI_DEFINE_MATH_OPERATORS
 #include "imgui_internal.h"
 
-#include <algorithm>
 
-inline bool ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
+#define PATH_SEP '/'
```

Comment:
This really needs something like:

#ifdef WIN32
#define PATH_SEP '\\'
#else
#define PATH_SEP '/'
#endif